isis/src/kaguya/objs/KaguyaTcCamera/KaguyaTcCamera.cpp

```diff
@@ -0,0 +1,160 @@
+ *   http://isis.astrogeology.usgs.gov, and the USGS privacy and disclaimers on
+ *   http://www.usgs.gov/privacy.html.
+ */
+#include "KaguyaTcCamera.h"
+#include "KaguyaTcCameraDistortionMap.h"
+
+#include <QString>
+
+#include "LineScanCameraDetectorMap.h"
+#include "CameraDistortionMap.h"
+#include "CameraFocalPlaneMap.h"
+#include "LineScanCameraGroundMap.h"
+#include "LineScanCameraSkyMap.h"
+#include "IException.h"
+#include "IString.h"
+#include "iTime.h"
+#include "NaifStatus.h"
+
+using namespace std;
+namespace Isis {
+  /**
+   * Constructor for the Kaguya TC Camera Model
+   *
+   * @param lab Pvl Label to create the camera model from
+   *
+   * @internal
+   *   @history 2018-10-02 Adam Goins & Jeannie Backer - Original Version
+   */
+  KaguyaTcCamera::KaguyaTcCamera(Cube &cube) : LineScanCamera(cube) {
+    m_instrumentNameLong  = "Terrain Camera";
+    m_instrumentNameShort = "TC";
+    m_spacecraftNameLong  = "Kaguya";
+    m_spacecraftNameShort = "Kaguya";
+
+    NaifStatus::CheckErrors();
+    // Get the camera characteristics
+    SetFocalLength();
+    SetPixelPitch();
+
+    // Get the start time in et
+    Pvl &lab = *cube.label();
+    PvlGroup inst = lab.findGroup("Instrument", Pvl::Traverse);
+
+    QString clockCount = inst["SpacecraftClockStartCount"];
+    double time = getClockTime(clockCount, -1, true).Et();
+
+    // Setup detector map
+    double lineRate = (double) inst["LineSamplingInterval"] / 1000.0;
+    
+    // Convert between parent image coordinates and detector coordinates (detector coordinate line, detector coordinate sample)   
+    LineScanCameraDetectorMap *detectorMap = new LineScanCameraDetectorMap(this, time, lineRate);
+
+    // Detetermine what to set the starting detector sample to, based on swath mode
+    QString swathMode = inst["SwathModeId"];
+
+    double startingDetectorSample = 1; 
+    if (swathMode.compare("FULL") == 0) {
+      startingDetectorSample = 1; 
+    }
+    else if (swathMode.compare("NOMINAL") == 0) {
+      startingDetectorSample = 297; 
+    }
+    else if (swathMode.compare("HALF") == 0) {
+      startingDetectorSample = 1172;
+    }
+
+    detectorMap->SetStartingDetectorSample(startingDetectorSample);
+    
+    CameraFocalPlaneMap *focalMap = new CameraFocalPlaneMap(this, naifIkCode());
+
+    // This is the same, no matter the swath mode
+
+    // This set the origin of the detector (not image samp,line). It is zero bassed.
+    // The detector offsets are 0,0 because the borsight is in the center of the array
+    QString key;
+    key = "INS" + toString(naifIkCode()) + "_BORESIGHT_SAMPLE";
+    double sampleBoreSight = getDouble(key);
+
+    key = "INS" + toString(naifIkCode()) + "_BORESIGHT_LINE";
+    double lineBoreSight = getDouble(key);
+    focalMap->SetDetectorOrigin(sampleBoreSight, lineBoreSight); 
+
+    // Setup distortion map
+    new KaguyaTcCameraDistortionMap(this, naifIkCode());
+//    new CameraDistortionMap(this);
```

Comment:
Is this not needed?

Reply:
Thanks for catching this. This was left in for debug and should indeed be removed.